File: Domain/Product.cs
using Domain.Model;

namespace Domain
{
    public class Product : BaseDeletableModel<Guid>
    {
        public Product()
        {
            this.Id = Guid.NewGuid();
        }
        public string Name { get; set; }

        public string Categoy { get; set; }

        public double Quantity { get; set; }

        public Guid UnitId { get; set; }
        public virtual Unit Unit { get; set; }
        
        public string Description { get; set; }

        public decimal Price { get; set; }

        public decimal DeliveryPrice { get; set; }
    }
}

File: client-app/src/app/models/product.ts
import { Unit } from "./unit"

export interface Product {
    name: string
    category: string
    quantity: number
    unitId: string
    unitAcronym: string
    description: string
    price: string
    deliveryPrice: string
    isDeleted: boolean
    id: string
    createdOn: Date | null
    modifiedOn: Date | null
    deletedOn: Date | null
    unitDto: Unit
}

export interface FormattedProduct extends Product {
    priceFormatted: string;
    deliveryPriceFormatted: string;
}

export interface UploadedProduct {
    id: string
    name: string
    category: string
    quantity: number
    unitId: string
    description: string
    price: string
    deliveryPrice: string
}